internal/invocation/stripe_executor.go

```diff
@@ -0,0 +1,95 @@
+package invocation
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+var (
+	// Default values differ from java impl. Also queue size is calculated differently.
+	// Java Client: queueSize per worker = defaultEventQueueCapacity / defaultEventWorkerCount
+	// Go Client: queueSize per worker = defaultEventQueueCapacity
+	defaultEventQueueCapacity = int32(10000)
+	defaultEventWorkerCount   = int32(runtime.NumCPU())
+)
+
+// executor represents the function that will run on workers of stripeExecutor.
+type executor func(queue chan func(), quit chan struct{}, wg *sync.WaitGroup)
+
+// stripeExecutor executes given "tasks" preserving the order among the ones that are given with the same key.
+type stripeExecutor struct {
+	quit       chan struct{}
+	wg         *sync.WaitGroup
+	execFn     executor
+	taskQueues []chan func()
+	queueCount int32
```

Comment:
Is there a reason why `queueCount` is `int32` instead of `int`?

Reply:
Will do a type conversion on invocation.PartitionID.

Reply:
I am moving this type conversion to `InvocationService.handleMessage()` since we need to it somewhere, because `invocation.PartitionID()` return "int32".
You are right this is cleaner